principles.rst

```diff
@@ -303,3 +309,130 @@ down GHC even when those features are not active. Yet this principle is importan
+  * Language editions like ``GHC202x`` should use only Stable extensions.
+
+* (GR1) is restricted to non-experimental extensions, and excludes dependence on ``ghc-experimental``.  That leaves the "experimental" space open to more rapid change.
+
+* For new language extensions, (GR1) is trivially satisfied if the change is gated behind an extension flag.
+
+* (GR1) is certainly broken if, say, we turn a warning into an error.
+
+
+* We will need to enumerate the "experimental features" mentioned above.
+
+Exceptions (GR2)
+~~~~~~~~~~~~~~~~
+
+**General rule (GR2)**.   *We may break (GR1), but only with a compelling reason, and with careful consideration of impact.*
+
+Compelling reasons include
+
+* To fix a security loophole.
+* To fix an outright bug in GHC.  It is possible that some code might accidentally rely on that bug; but we can't use that as a reason to grandfather the bug indefinitely.  (Imagine that 2+77 = 80.)  There is a judgement call here, about what a "bug" is.
+* To fix a design flaw in the language.   The switch to simple subsumption is an example of such a change, albeit one that was not well handled at the time.
+
+This list is not exhaustive, but the emphasis is on "compelling reason", bearing in mind the costs that any change imposes on users.
+
+Fixing an outright bug, or correcting (what we now consider to be) a design flaw in the language, counts as "compelling" reasons, even though they threaten stability.
+
+* We want Haskell to be a beautiful, elegant language, not one riddled with inconsistencies, grandfathered in simply because fixing the inconsistency would risk some breakage.
+* Similarly, we acknowledge the possibility of making major changes, if we truly believe that they will make the language better, and/or open the door to other improvements.
+* Yet we should not make such changes lightly, because they impose real costs.
+* Choices should be informed by the amount of breakage, e.g. by compiling thousands of packages in Hackage.  Changes that break little or nothing need a much less compelling reason than changes that break a lot.
+* Changes, especially significant changes, should be introduced gradually: the subject of (GR3).
+
+Deprecations (GR3)
+~~~~~~~~~~~~~~~~~~~~
+
+**General rule (GR3)**.  *If we break (GR1), for a compelling reason (GR2), we should whenever possible provide a deprecation cycle, as well as copious publicity, so that users are alerted (via depreciation warnings) to the upcoming change, have time to adapt, and can raise concerns.*
+
+A deprecation cycle should give at least one GHC major release for authors to make changes.  In the case of changes that are more difficult to accommodate, we may consider a second cycle.  (Side note: there is a `separate conversation going on <https://github.com/haskell/pvp/issues/58>`_ about deprecation warnings and the PVP.)
```

Comment:
PVP 1.1 (not released yet) already has revised point 7 of PVP, removing the need (or rather recommendation) to do a major PVP bump for a mere deprecation.

So there's no discussion needed for that part anymore. It was already decided.